src/main/java/uk/ac/ebi/embl/template/CSVReader.java

```diff
@@ -14,63 +14,66 @@
-            }
+            return processTemplateSpreadsheetLine();
+        }
+        return null;
+    }
 
-            String entryNumber = currentTokenLine[0];
-            if (entryNames.contains(entryNumber.toUpperCase())) {
-                throw new TemplateUserError(CSVWriter.HEADER_TOKEN + " must be unique. "+ currentTokenLine[0] + " exists more than once" );
-            } else {
-                entryNames.add(entryNumber.toUpperCase());
-            }
-            entryTokensMap.setSequenceName(entryNumber);
+    public void skipEmptyLinesAndComments() throws Exception {
+        // Skip empty lines and comment lines.
+        while(currentLine != null && (
+              currentLine.isEmpty() ||
+              currentLine.startsWith(FastaSpreadsheetConverter.COMMENT_TOKEN))) {
+            readLine();
+        }
+    }
 
-            for (int i = 1; i < currentTokenLine.length; i++) {
-                String tokenValue = currentTokenLine[i];
-                checkTokenForBannedCharacters(tokenValue);
-                tokenValue = tokenValue.replaceAll("<br>", "\n");
-                tokenValue = tokenValue.replaceAll(";", ",");
-                if (tokenValue.startsWith("\"") && tokenValue.endsWith("\"")) {
-                    tokenValue = StringUtils.stripStart(tokenValue, "\"");
-                    tokenValue = StringUtils.stripEnd(tokenValue, "\"");
-                }
-                entryTokensMap.addToken(headerKeys.get(i), tokenValue);
+    public CSVLine processTemplateSpreadsheetLine() throws Exception {
+        final TemplateVariables entryTokensMap = new TemplateVariables();
+        final String[] currentTokenLine = StringUtils.splitPreserveAllTokens(currentLine, CSVWriter.UPLOAD_DELIMITER);
+        if ((currentTokenLine.length) != headerKeys.size()) {
+            String lineSummary = currentLine;
+            if (currentLine.length() > 10)
+                lineSummary = currentLine.substring(0, 10);
+            throw new TemplateUserError("There are " + headerKeys.size() + " tokens specified in the header but " + currentTokenLine.length + " values for entry on line " + lineSummary + "..., please check your import file data is properly delimited with a 'tab'.");
+        }
+
+        // The entry number will be used as the sequence name and must be unique
+        // for a submission.
+        entryNumber++;
+        if (entryNumbers.contains(entryNumber)) {
+            throw new TemplateUserError("Internal error: non-unique sequence number error" );
+        }
+        entryNumbers.add(entryNumber);
+
+        entryTokensMap.setSequenceName( String.valueOf(entryNumber));
+
+        for (int i = 1; i < currentTokenLine.length; i++) {
+            String tokenValue = currentTokenLine[i];
+            checkTokenForBannedCharacters(tokenValue);
```

Comment:
This is a bug: Since ENTRYNUMBER is not entryTokensMap the for loop must iterate from "0". 
Hope we must update the test files to remove ENTRYNUMBER.